treeherder/perf/models.py

```diff
@@ -81,9 +84,45 @@ def __str__(self):
                                     self.last_updated)
 
 
+class PerformanceDatumManager(models.Manager):
+    """
+    Convenience functions for operations on groups of performance datums
+    """
+
+    def cycle_data(self, repository, cycle_interval, chunk_size, sleep_time):
+        """Delete data older than cycle_interval, splitting the target data
+into chunks of chunk_size size."""
+
+        max_timestamp = datetime.datetime.now() - cycle_interval
+
+        # seperate datums into chunks
+        perf_datums_to_cycle = list(self.filter(
+            repository=repository,
+            push_timestamp__lt=max_timestamp).values_list('id', flat=True))
+        chunk_list = zip(*[iter(perf_datums_to_cycle)] * chunk_size)
+        # append the remaining job data not fitting in a complete chunk
+        chunk_list.append(
+            perf_datums_to_cycle[-(len(perf_datums_to_cycle) % chunk_size):])
+
+        for chunk in chunk_list:
+            self.filter(id__in=chunk).delete()
+            if sleep_time:
+                # Allow some time for other queries to get through
+                time.sleep(sleep_time)
+
+        # also remove any signatures which are (no longer) associated with
+        # a job
+        for signature in PerformanceSignature.objects.filter(
```

Comment:
This is only 84 characters on one line -- up to 100 characters I'm fine with not wrapping :-)